src/app/client/src/app/modules/observation/service/observation-util/observation-util.service.ts

```diff
@@ -0,0 +1,170 @@
+import { Injectable } from "@angular/core";
+import { UserService } from "@sunbird/core";
+import { IUserData, ConfigService,ResourceService } from "@sunbird/shared";
+import { take } from "rxjs/operators";
+import { KendraService } from "@sunbird/core";
+import { AlertModal } from "../../components/alert-modal/alert-modal.component";
+import {SuiModalService} from "ng2-semantic-ui";
+import { Router } from "@angular/router";
+
+
+@Injectable({
+  providedIn: "root",
+})
+export class ObservationUtilService {
+  config;
+  requiredFields;
+  profile: any;
+  dataParam: any;
+  constructor(
+    public userService: UserService,
+    config: ConfigService,
+    private kendraService: KendraService,
+    public modalService: SuiModalService,
+    public resourceService: ResourceService,
+    public router: Router
+  ) {
+    this.config = config;
+  }
+
+  getProfileData() {
+    return this.userService.userData$
+      .pipe(take(1))
+      .toPromise()
+      .then((profileData: IUserData) => {
+        if (
+          profileData &&
+          profileData.userProfile &&
+          profileData.userProfile["profileUserType"] &&
+          profileData.userProfile["profileUserType"]["subType"] === null
+        ) {
+          return false;
+        } else {
+          return true;
+        }
+      })
+      .catch((error) => {});
+  }
+
+  async getMandatoryEntities(): Promise<any> {
+    const profile = JSON.parse(
```

Comment:
I'd suggest you add try catch block here. Whatif your localStorage item isn't present, then code would break. Else add a catch block at the entry function.